### PR TITLE
Support Elixir 1.13

### LINKED ISSUE #4019: Compatibility with Elixir 1.13
Currently using 1.13.2 to build there is the following issue:

```
* rabbit_common (/home/bakkenl/development/rabbitmq/rabbitmq-server/deps/rabbit_common)
  could not find an app file at "_build/dev/lib/rabbit_common/ebin/rabbit_common.app". This may happen if the dependency was not yet compiled or the dependency indeed has no app file (then you can pass app: false as option)
** (Mix) Can't continue due to errors on dependencies
```

Discussion: https://groups.google.com/g/rabbitmq-users/c/cB_pG6TqBDQ

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #4019

This is the build error prior to these changes:

```
* rabbit_common (/home/bakkenl/development/rabbitmq/rabbitmq-server/deps/rabbit_common)
  could not find an app file at "_build/dev/lib/rabbit_common/ebin/rabbit_common.app". This may happen if the dependency was not yet compiled or the dependency indeed has no app file (then you can pass app: false as option)
** (Mix) Can't continue due to errors on dependencies
```

Telling `mix` to compile `rabbit_common` ensures that the following
links are created:

```
$ ll deps/rabbitmq_cli/_build/dev/lib/rabbit_common/
total 8
drwxr-xr-x  2 bakkenl bakkenl 4096 Jan 20 09:46 .
drwxr-xr-x 10 bakkenl bakkenl 4096 Jan 20 09:46 ..
lrwxrwxrwx  1 bakkenl bakkenl   33 Jan 20 09:46 ebin -> ../../../../../rabbit_common/ebin
lrwxrwxrwx  1 bakkenl bakkenl   36 Jan 20 09:46 include -> ../../../../../rabbit_common/include
```

### CODE REVIEW

**Comment by pjk25** [comment]
Those must be the symlinks that result in problems under the windows build with bazel?

**Comment by lukebakken** [comment]
> Those must be the symlinks that result in problems under the windows build with bazel?

Hm, I'm not sure. I haven't tried the bazel/Windows build using Elixir 1.13.X.

**Comment by pjk25** [comment]
I will try removing `compile: false,` and see if that is the trick.

**Comment by pjk25** [comment]
Removing `compile: false` wasn't it, but maybe the solution in #4028 will work here as well?

File: deps/rabbitmq_cli/mix.exs [hunk 1]
```diff
@@ -149,19 +149,15 @@ defmodule RabbitMQCtl.MixfileBase do
...
           {
             :rabbit_common,
             path: Path.join(deps_dir, "rabbit_common"),
-            compile: false,
             override: true
           },
           {
             :amqp_client,
             path: Path.join(deps_dir, "amqp_client"),
-            compile: false,
```
Comment (pjk25) [inline deps/rabbitmq_cli/mix.exs:164]: @lukebakken Given the changes to `deps/rabbitmq_cli/rabbitmqctl.bzl` in #4028, I suspect it is no longer necessary to remove these lines.

**Comment by pjk25** [comment]
@lukebakken Currently our actions workflows don't run with the latest elixir (https://github.com/rabbitmq/rbe-erlang-platform/blob/3e3b122a363c2fe83da4089e86caf7c03dee4b29/.github/workflows/rbe_configs_gen.yaml#L14-L15). Do we want to consider changing that?

**Comment by michaelklishin** [comment]
@pjk25 we can use the latest if it's available on Actions. Our release builds use Debian packages so that would have to be updated to 1.13, too.

**Comment by michaelklishin** [comment]
Note that we can allow 1.13 to be used regardless. Many core team members will run Elixir 1.13 soon enough.

**Comment by michaelklishin** [comment]
Lastly, as far as I can tell there are no Erlang/OTP version requirement changes in 1.13. Which simplifies things for us.

**Comment by michaelklishin** [comment]
I found a minor breaking change that affects `test/core`, looking into it.

**Comment by michaelklishin** [comment]
In Elixir `1.13`, an `OptionParser` switch cannot be of type `:atom`. Arguably [this has not been a documented option in `1.12` either](https://hexdocs.pm/elixir/1.12/OptionParser.html). Fortunately we can easily work around this.

**Comment by michaelklishin** [comment]
Will re-submit.

**Pull request closed without merging**